pkg/uid/generator.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// For how the uidshift and uidcount are generated please check:
+// http://cgit.freedesktop.org/systemd/systemd/commit/?id=03cfe0d51499e86b1573d1
+
+package uid
+
+// Generator defines the interface for generating a uid/gid.
+type Generator interface {
```

Comment:
You shouldn't be able to get one and not the other, but the only offender in this case is where you get UID from passwd and GID from group. Can you add a function to passwd to get GID from file? If you need me to do it, let me know.